README.md

```diff
@@ -1,173 +1,175 @@
-* `"exitproof"(erdstall.BalanceProof)`:
-		triggered when a balance proof intended for exiting Erdstall on-chain is received from a proof subscription.
-	This happens after a call to `Erdstall.exit()` or during `Erdstall.leave()`.
-
-The following `ErdstallEvent` smart contract events exist:
-
-* **`"Deposited"`**:
-		you deposited money into Erdstall.
-* **`"Withdrawn"`**:
-		you withdrew funds from Erdstall.
-* `"Frozen"`:
-		an unrecoverable error occurred and the Erdstall system is now frozen forever, and funds must be withdrawn using the latest "sealed" balance proof.
-* `"Challenged"`:
-		someone challenged the Erdstall operator to prove it is still alive.
-* `"ChallengeResponded"`:
-		the Erdstall operator successfully proved it still operates.
-* `"TokenTypeRegistered"`:
-		a new token type (such as ETH, ERC20, ERC721, etc.) has been registered with Erdstall on-chain.
-* `"TokenRegistered"`:
-		a specific token contract has been registered on-chain.
-
-
-**Finishing up**&emsp;<!--
--->After all events have been set up, call `Erdstall.initialize()` to establish the client's connection to the Erdstall ledger.
-You're now good to go!
-The `open` event will be triggered and your logic will launch.
+TypeScript declaration files are shipped with the npm package. Full [TypeDoc](https://typedoc.org/)-generated documentation of the source code can be found at [this repository's Github pages](https://perun-network.github.io/erdstall-ts-sdk/).
+
+## Erdstall interactions
+
+The SDK's primary goal is to facilitate interaction with an Erdstall off-chain network as well as the on-chain Erdstall smart contract. The following describes the most important functions of `ErdstallClient`s (read-only) and `ErdstallSession`s (also writing).
+
+### Account queries
+
+The current balance of any account can be queried using `Erdstall.getAccount(who)`.
+
+### Entering and leaving
+
+Fungible Tokens and NFTs can be deposited into the Erdstall network by calling
+`ErdstallSession.deposit(assets)`. This call returns a list of promises of on-chain transactions. Only after all those transactions are mined are the deposits successful. The assets will be available in Erdstall once the blocks containing them have reached the enclave and sufficient finality depth has been reached.
+
+To exit an Erdstall network with all assets, call `ErdstallSession.leave()`. This will first send an `exit` request to the Erdstall enclave. Once the epoch has ended, an exit proof will be received from the operator. Due to protocol-security-related reasons, it now has to be waited for another full epoch before an on-chain `withdraw` transaction can be sent to the Erdstall contract, containing the exit proof that was received in the prior epoch. The call returns a list of transaction promises. Once all those transactions have been mined, the user's assets will be withdrawn from Erdstall back into their Ethereum wallet.
```

Comment:
Do we have to explain this concept?

Reply:
Which concept exactly?

Reply:
The concept of an epoch, but probably people who use the sdk should be aware.

Reply:
Decided to do a glossary as a separate issue, possibly not in the README file. https://github.com/perun-network/erdstall-ts-sdk/issues/122